pkg/client/cache/reflector.go

```diff
@@ -285,7 +287,11 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			return nil
 		}
 		if err := r.watchHandler(w, &resourceVersion, resyncCh, stopCh); err != nil {
-			if err != errorResyncRequested && err != errorStopRequested {
+			if err == errorResyncRequested {
+				glog.V(1).Infof("%s: watch of %v ending because handler requested resync", r.name, r.expectedType)
+			} else if err == errorStopRequested {
+				glog.V(1).Infof("%s: watch of %v ending because handler requested stop", r.name, r.expectedType)
+			} else {
```

Comment:
SGTM since these shouldn't be happening regularly, suggest just an || and logging the error itself but leave it to you